tests/integration/test_styles_sdclient.py

```diff
@@ -124,12 +123,7 @@ def test_class_name_matches_css_object_name(mocker, main_window):
     assert "ReplyWidget" in error_message.objectName()
 
 
-def test_class_name_matches_css_object_name_for_print_dialog(print_dialog):
-    assert "PrintDialog" == print_dialog.__class__.__name__
-
-
-def test_class_name_matches_css_object_name_for_export_dialog(export_dialog):
-    assert "ExportDialog" == export_dialog.__class__.__name__
+def test_preferred_name_matches_css_object_name_for_export_dialog(export_dialog):
     assert "ExportDialog" in export_dialog.passphrase_form.objectName()
```

Comment:
These two specific assertions are not met anymore. I decided not to touch the stylesheet in order to keep the changes scoped (they're large already :grimacing:).
Eventually, I think that breaking the stylesheet into smaller files that follow the package structure would make maintenance easier, and also would make redundant the need to have global matching names for Python classes and QSS names.

Until then, `PrintDialog` and `ExportDialog` seems to me that they remain the most appropriate QSS names for `conversation.ExportDialog` and `conversation.PrintDialog` (even though they are called `export.Dialog` and `print.Dialog` as an implementaiton detail of the `conversation` package.) Does this make sense?

Reply:
The convention, enforced by these tests, is to use the class name as the object name in the CSS to make it easier to find the relevant code. For now, let's stick to this convention and rename the CSS objects to match the new class names. This will make it cleaner when you're ready to move the relevant CSS code to the new project structure.

Reply:
The direction of this PR is great! I just have one main concern that this breaks a couple conventions:
* tests should exist in a test structure that mirrors the project structure, e.g. tests that test `securedrop_client/gui/dialogs.py` need to go in `tests/gui/dialogs.py`
* CSS object names should match the Python classes they are used in

If you refactor one class at a time, you can migrate the relevant tests so that you don't break these conventions, and it's still small enough for the reviewer.

Were you thinking of doing something like this eventually?

```
styles/gui
├── auth
│   └── dialog.css
├── buttons.css
├── conversation
│   └── print
│       └── dialog.css
├── dialogs.css
└── ...

tests/gui
├── test_init.py
├── auth
│   ├── test_init.py
│   └── test_dialog.py
├── test_buttons.py
├── conversation
│   ├── test_init.py
│   ├── export
│   │   ├── test_init.py
│   │   └── test_dialog.py
│   ├── test_file.py
│   └── print
│       ├── test_init.py
│       └── test_dialog.py
├── test_dialogs.py
└── ...
```
Or would you move the css files into the project structure like this:
```
securedrop_client/gui
├── __init__.py
├── auth
│   ├── __init__.py
│   └── dialog.py
│   └── dialog.css
├── buttons.py
├── buttons.css
├── conversation
│   ├── __init__.py
│   ├── export
│   │   ├── __init__.py
│   │   └── dialog.py
│   ├── file.py
│   └── print
│       ├── __init__.py
│       └── dialog.py
├── dialogs.py
├── dialogs.css
└── ...
```
Or keep it flat:
```
styles
├── auth_dialog.css
├── print_dialog.css
└── dialogs.css
```

Reply:
I can add those changes :+1: 

Regarding the stylesheets location, I have a strong preference for the second layout you described, where the all the parts of a given component are grouped together, for a few resaons:

- That option splits concerns by _component_, which in our case closely aligns with the domain boundaries (e.g. `auth`, `conversation`, `source`, "conversation export feature", "source deletion feature" are concepts that are shared by the team _beyond_ the code.
- That option also makes clear the fact that components are scoped to themselves. Because all the files are grouped together it makes less likely that dependencies will be drawn from unrelated places.
- I find the separation by language just as arbitrary and not very convincing. After all, QSS and Python code do indeed change together (we even have tests to check object names and QSS names are in sync!). ([Support material](https://speakerdeck.com/didoo/let-there-be-peace-on-css?slide=62) - that entire side deck is insteresting IMHO) 

What do you think @creviera? (and @rocodes @Er1kkkaaa!)

Note: this thread is relevant to https://github.com/freedomofpress/securedrop-client/issues/303

Reply:
I agree with your points. :rocket: 

Since there's a lot of code moving around, it might make sense to break this up into multiple commits or PRs scoped to moving one class/file at a time, e.g. move `ModalDialog` and any associated css and tests from `widgets.py`, `test_widgets.py`, and some-name.css into `dialogs.py`, `test_dialogs.py`, and `dialog.css`. This should make both of our lives much easier during the migration and review of the migration. Otherwise, I'm down for a slower review where I piece it all together in the diffs.

Reply:
> Since there's a lot of code moving around, it might make sense to break this up into multiple commits or PRs scoped to moving one class/file at a time [...]

Yes! I was thinking the same, I'll "re-package" this PR (and #1373) into one-class-at-a-time chunks. :+1:

Reply:
For road mapping purposes, I'll proceed as follows:

- [ ] `misc` (removing all class definitions from `gui/__init__.py`)
- [ ] `actions` (made possible by :point_up: if I remember correctly)
- [ ] `auth` (currently is the smaller package)
- [ ] `conversation/file`
- [ ] `conversation/export`
- [ ] `conversation/print`

I'm not 100% sure about the end of the sequence, but I'll update when we get there.